content/en/docs/tasks/observability/metrics/using-istio-dashboard/test.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2154,SC2155,SC2034
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+source "tests/util/helpers.sh"
+source "tests/util/samples.sh"
+source "tests/util/addons.sh"
+
+# @setup profile=default
+
+_deploy_and_wait_for_addons prometheus grafana
+_verify_like snip_viewing_the_istio_dashboard_1 "$snip_viewing_the_istio_dashboard_1_out"
+_verify_like snip_viewing_the_istio_dashboard_2 "$snip_viewing_the_istio_dashboard_2_out"
+
+# TODO: Deploying bookinfo and sending requests through ingress gateway
+# seems like a common pattern for many tests. Should be moved to tests/util. 
+kubectl label namespace default istio-injection=enabled --overwrite
+startup_bookinfo_sample
+_set_ingress_environment_variables
+GATEWAY_URL="$INGRESS_HOST:$INGRESS_PORT"
+bpsnip_trace_generation__1
+
+snip_viewing_the_istio_dashboard_3 &
+
+# For verification, we only check if the dashboards are accessible, but not its actual contents
+# TODO: Is it worth checking API calls and output for Grafana case? 
```

Comment:
Yes, similar to my comment for jaeger, above, I think you should generate some traffic (as the doc describes), and then do some minimal check to confirm that something was generated. We don't need to confirm it's exactly correct, leave that to the istio/istio integration tests, but just do a basic sanity check to confirm that logging did happen as a result of the generated traffic.

Reply:
I added some checks. I had to dig into the queries used by Grafana to contact Prometheus to populate the dashboard. It is long and ugly. I have added it anyways.